envs/env_4inrow.py

```diff
@@ -18,39 +18,61 @@ def __init__(self, player1, player2, board_size):
         self.player2 = player2
         self.boardH = board_size[0]
         self.boardW = board_size[1]
-        self.board = [[0] * self.boardW] * self.boardH
+        self.board = [[0 for i in range(self.boardW)] for j in range(self.boardH)]
 
     def reset(self):
-        self.board = [[0] * self.boardW] * self.boardH
+        self.board = [[0 for i in range(self.boardW)] for j in range(self.boardH)]
         return self.board
 
     def apply_action(self, player, action):
-        if action in self.available_moves(player):
-            self.board[action[0]][action[1]] = player
-            # next state, reward
-            return self.board, 0
-        return None
+        for a in self.available_moves(player):
+            if action == a[1]:
```

Comment:
What is `a[1]`? Why not `a[0]`? I thought `for a in self.available_moves(player)` returns the available moves, shouldn't `a` by itself be the available action?

Reply:
a is a tuple, which actin is an int, representing the column. a[1] is the column, which I check against.

Reply:
Then why wouldn't you just return a list of ints from `self.available_moves` ?